spec/integration/blocked_hotlinked_image_spec.rb

```diff
@@ -0,0 +1,138 @@
+# frozen_string_literal: true
+
+describe "hotlinked image blocking" do
+  let(:hotlinked_url) { "http://example.com/images/2/2e/Longcat1.png" }
+  let(:onebox_url) { "http://example.com/onebox" }
+  let(:png) { Base64.decode64("R0lGODlhAQABALMAAAAAAIAAAACAAICAAAAAgIAAgACAgMDAwICAgP8AAAD/AP//AAAA//8A/wD//wBiZCH5BAEAAA8ALAAAAAABAAEAAAQC8EUAOw==") }
+
+  before do
+    SiteSetting.download_remote_images_to_local = false
+    stub_request(:get, hotlinked_url).to_return(body: png, headers: { "Content-Type" => "image/png" })
+    stub_image_size
+  end
+
+  it "normally allows hotlinked images" do
+    post = Fabricate(:post, raw: "<img src='#{hotlinked_url}'>")
+    expect(post.cooked).to have_tag("img", with: { "src" => hotlinked_url })
+  end
+
+  context "with hotlinked media blocked, before post-processing" do
+    before do
+      SiteSetting.block_hotlinked_media = true
+      Oneboxer.stubs(:cached_onebox).returns("<aside class='onebox'><img src='#{hotlinked_url}'></aside>")
+    end
+
+    it "blocks hotlinked images" do
+      post = Fabricate(:post, raw: "<img src='#{hotlinked_url}'>")
+      expect(post.cooked).not_to have_tag("img[src]")
+      expect(post.cooked).to have_tag("img", with: { PrettyText::BLOCKED_HOTLINKED_SRC_ATTR => hotlinked_url })
+    end
+
+    it "blocks hotlinked videos with src" do
+      post = Fabricate(:post, raw: "![alt text|video](#{hotlinked_url})")
+      expect(post.cooked).not_to have_tag("video source[src]")
+      expect(post.cooked).to have_tag("video source", with: { PrettyText::BLOCKED_HOTLINKED_SRC_ATTR => hotlinked_url })
+    end
+
+    it "blocks hotlinked videos with srcset" do
+      srcset = "#{hotlinked_url} 1x,https://example.com 2x"
+      post = Fabricate(:post, raw: "<video><source srcset='#{srcset}'></video>")
+      expect(post.cooked).not_to have_tag("video source[src]")
```

Comment:
```suggestion
      expect(post.cooked).not_to have_tag("video source[srcset]")
```

Should this be `srcset` instead of `src`?